services/billing.py

```diff
@@ -237,16 +237,31 @@ def modify_subscription(self, owner, desired_plan):
                 ],
                 metadata=self._get_checkout_session_and_subscription_metadata(owner),
                 proration_behavior=proration_behavior,
+                payment_behavior="pending_if_incomplete",
             )
-
-            plan_service = PlanService(current_org=owner)
-            plan_service.update_plan(
-                name=desired_plan["value"], user_count=desired_plan["quantity"]
-            )
-
             log.info(
-                f"Stripe subscription modified successfully for owner {owner.ownerid} by user #{self.requesting_user.ownerid}"
+                f"Stripe subscription upgrade attempted for owner {owner.ownerid} by user #{self.requesting_user.ownerid}"
             )
+            indication_of_payment_failure = getattr(
+                subscription, "pending_update", None
+            )
+            if indication_of_payment_failure:
+                # payment failed, raise this to user by setting as delinquent
+                owner.delinquent = True
+                owner.save()
+                log.info(
+                    f"Stripe subscription upgrade failed for owner {owner.ownerid} by user #{self.requesting_user.ownerid}",
+                    extra=dict(pending_update=indication_of_payment_failure),
+                )
+            else:
+                # payment successful
```

Comment:
will delinquent ever be true at this point where we'd want to set to false?

Reply:
love this suggestion, but "yes and"ed it with this change https://github.com/codecov/codecov-api/pull/831/files#diff-dda355f2f33801fe525e8ba51493017a8957e35710e9dc1d4affed3940e18685R56 (clearing delinquency in `update_plan`!)
From what I could tell, we are only using `plan_service.update_plan` in a situation where the `owner` should not be `delinquent`, so just make sure that field is set correctly while we're updating the owner obj.